frontend/src/components/layout/topmessage/HolidayPromoBanner.tsx

```diff
@@ -0,0 +1,72 @@
+import Link from "next/link";
+import { useL10n } from "../../../hooks/l10n";
+import { Localized } from "../../Localized";
+import styles from "./HolidayPromoBanner.module.scss";
+import { RuntimeData } from "../../../hooks/api/runtimeData";
+import {
+  getPeriodicalPremiumSubscribeLink,
+  isPeriodicalPremiumAvailableInCountry,
+} from "../../../functions/getPlan";
+import { ProfileData } from "../../../hooks/api/profile";
+import { useEffect, useState } from "react";
+
+type Props = {
+  profile?: ProfileData;
+  runtimeData?: RuntimeData;
+};
+
+export const HolidayPromoBanner = (props: Props) => {
+  const l10n = useL10n();
+  const [loadingProfile, setLoadingProfile] = useState(true);
+  const coupon = "HOLIDAY20";
+  const couponExpiry = "Dec 31, 2023";
+  const subscribeLink = isPeriodicalPremiumAvailableInCountry(props.runtimeData)
+    ? getPeriodicalPremiumSubscribeLink(props.runtimeData, "yearly")
+    : null;
+
+  useEffect(() => {
+    setLoadingProfile(false);
+  }, [props.profile]);
```

Comment:
It's probably fine, because I think this works, but it feels kinda weird to have this. We're passing in `profile.data`, but [there's also `profile.isLoading`](https://swr.vercel.app/docs/api#return-values) which sounds like it's made for this use case. But fine to leave as-is.

Reply:
Actually I realized the useEffect doesn't seem to solve the problem, but this isLoading property works, thanks for mentioning it!